onnxruntime/core/providers/dml/DmlExecutionProvider/src/ExecutionProvider.cpp

```diff
@@ -526,14 +526,30 @@
         std::string partitionKernelPrefix = std::to_string(m_partitionKernelPrefixVal++) + "_";
         uint32_t deviceDataTypeMask = GetSupportedDeviceDataTypeMask();
 
-        return PartitionGraph(
-            graph,
-            *m_internalRegInfoMap,
-            registries,
-            deviceDataTypeMask,
-            m_kernelRegistry.get(),
-            partitionKernelPrefix
-        );
+        bool isFirstPartitionCall = std::none_of(graph.Nodes().begin(), graph.Nodes().end(), [](const onnxruntime::Node& node) {
+                return !node.GetExecutionProviderType().empty() && node.GetExecutionProviderType() == onnxruntime::kDmlExecutionProvider;
```

Comment:
>!node.GetExecutionProviderType().empty() [](http://example.com/codeflow?start=23&length=40)

The `empty` check is redundant, because if empty, then type != DML anyway, and if not empty, then only the comparison matters. So, I'd just delete this clause.

Reply:
Make sense. I have added it just to be extra cautious.